source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -37,72 +42,89 @@ const Http::HeaderEntry *getHeader(const Http::HeaderMap &header_map,
 } // namespace
 
 Extractor::Extractor(const envoy::api::v2::filter::http::Extraction &extractor)
-    : headername_(extractor.header()), group_(extractor.subgroup()),
+    : headername_(extractor.header()), body_(extractor.has_body()),
+      group_(extractor.subgroup()),
       extract_regex_(Regex::Utility::parseStdRegex(extractor.regex())) {}
-std::string Extractor::extract(const Http::HeaderMap &header_map) const {
+
+absl::string_view Extractor::extract(const Http::HeaderMap &header_map,
+                                     GetBodyFunc body) const {
   // TODO: should we lowercase them in the config?
-  const Http::HeaderEntry *header_entry = getHeader(header_map, headername_);
-  if (!header_entry) {
-    return "";
+  if (body_) {
+    return extractValue(body());
+  } else {
+    const Http::HeaderEntry *header_entry = getHeader(header_map, headername_);
+    if (!header_entry) {
+      return "";
+    }
+    return extractValue(header_entry->value().getStringView());
   }
+}
 
-  std::string value(header_entry->value().getStringView());
-
+absl::string_view Extractor::extractValue(absl::string_view value) const {
   // get and regex
-  std::smatch regex_result;
-  if (std::regex_match(value, regex_result, extract_regex_)) {
-    std::smatch::iterator submatch_it = regex_result.begin();
-    for (unsigned i = 0; i < group_; i++) {
-      std::advance(submatch_it, 1);
-      if (submatch_it == regex_result.end()) {
-        return "";
-      }
+  std::match_results<absl::string_view::const_iterator> regex_result;
+  if (std::regex_match(value.begin(), value.end(), regex_result,
+                       extract_regex_)) {
+    if (group_ >= regex_result.size()) {
+      // TODO: this indicates user misconfiguration... log or stat?
```

Comment:
Why not log about this now? Or possibly check the number of matches in the constructor and panic there so can check the nack

Reply:
we actually can! i didnt have the callbacks before.
i will see if i can do the check in the ctor, good idea.